elytron/src/test/java/org/wildfly/extension/elytron/RoleMappersTestCase.java

```diff
@@ -156,4 +167,216 @@ public void testKeepBothMappedRoleMapper() throws Exception {
+        init("TestDomain9");
+
+        ServiceName serviceName = Capabilities.SECURITY_DOMAIN_RUNTIME_CAPABILITY.getCapabilityServiceName("TestDomain9");
+        Assert.assertNotNull(services.getContainer());
+        Assert.assertNotNull(services.getContainer().getService(serviceName));
+        SecurityDomain domain = (SecurityDomain) services.getContainer().getService(serviceName).getValue();
+        Assert.assertNotNull(domain);
+
+        ServerAuthenticationContext context = domain.createNewAuthenticationContext();
+        context.setAuthenticationName("user4");
+        Assert.assertTrue(context.exists());
+        Assert.assertTrue(context.authorize());
+        context.succeed();
+        SecurityIdentity identity = context.getAuthorizedIdentity();
+        Assert.assertEquals("user4", identity.getPrincipal().getName());
+
+        Roles roles = identity.getRoles();
+        Assert.assertFalse(roles.contains("app-user"));
+        Assert.assertFalse(roles.contains("app-user-first-time-user"));
+        Assert.assertFalse(roles.contains("app-admin-first-time-user"));
+        Assert.assertFalse(roles.contains("app-user-first-time-admin"));
+        Assert.assertFalse(roles.contains("joe"));
+        Assert.assertFalse(roles.contains("app-admin"));
+        Assert.assertFalse(roles.contains("app-admin-first-time-admin"));
+
+        context = domain.createNewAuthenticationContext();
+        context.setAuthenticationName("user7");
+        Assert.assertTrue(context.exists());
+        Assert.assertTrue(context.authorize());
+        context.succeed();
+        identity = context.getAuthorizedIdentity();
+        Assert.assertEquals("user7", identity.getPrincipal().getName());
+
+        roles = identity.getRoles();
+        Assert.assertTrue(roles.contains("admin"));
+        Assert.assertFalse(roles.contains("user"));
+    }
+
+    @Test
+    public void testAddRegexRoleMapperAggregate() throws Exception {
+        init("TestDomain10");
+
+        ServiceName serviceName = Capabilities.SECURITY_DOMAIN_RUNTIME_CAPABILITY.getCapabilityServiceName("TestDomain10");
+        Assert.assertNotNull(services.getContainer());
+        Assert.assertNotNull(services.getContainer().getService(serviceName));
+        SecurityDomain domain = (SecurityDomain) services.getContainer().getService(serviceName).getValue();
+        Assert.assertNotNull(domain);
+
+        ServerAuthenticationContext context = domain.createNewAuthenticationContext();
+        context.setAuthenticationName("user5");
+        Assert.assertTrue(context.exists());
+        Assert.assertTrue(context.authorize());
+        context.succeed();
+        SecurityIdentity identity = context.getAuthorizedIdentity();
+        Assert.assertEquals("user5", identity.getPrincipal().getName());
+
+        Roles roles = identity.getRoles();
+        Assert.assertTrue(roles.contains("admin"));
+        Assert.assertTrue(roles.contains("admin"));
```

Comment:
Maybe duplicated line here? From test domain definition I ca see that `user5` should have the following roles, resulting from `AggregateRoleMapper1` processing:
* `1-user-1`: match, translated into `admin`
* `joe`: no match, not kept
* `2-guest-2`: match, translated into `guest`

Hence looks like result should be `{admin, guest}`.
WDYT?

Reply:
fixed